### PR TITLE
8265756: AArch64: initialize memory allocated for locals according to Windows AArch64 stack page growth requirement in template interpreter

### PR DESCRIPTION
**Pull request by voitylov**
Hi,

please review this PR which fixes several JCK test failures on Windows AArch64. On Windows AArch64, the order in which the pages are touched during stack allocation, matters. Here is a quote: "Functions that allocate 4k or more worth of stack must ensure that each page prior to the final page is touched in order" [1]. Since on Linux and Mac the order does not matter, the suggested fix reverses the order in which memory pages are touched during stack allocation for locals in template interpreter for all platforms.

Testing: JCK, jtreg, newly developed regression test on Linux AArch64, Windows AArch64, Mac AArch64, Pre-submit tests (which includes the newly developed test).

[1] https://docs.microsoft.com/en-us/cpp/build/arm64-windows-abi-conventions?view=msvc-160#stack

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265756](https://bugs.openjdk.java.net/browse/JDK-8265756): AArch64: initialize memory allocated for locals according to Windows AArch64 stack page growth requirement in template interpreter

### Reviewers
 * [Andrew Dinn](https://openjdk.java.net/census#adinn) (@adinn - **Reviewer**)
 * [Andrew Haley](https://openjdk.java.net/census#aph) (@theRealAph - **Reviewer**) ⚠️ Review applies to 5e8c7d6a904cbbd0318151836421b83931f690c7

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3633/head:pull/3633` \
`$ git checkout pull/3633`

Update a local copy of the PR: \
`$ git checkout pull/3633` \
`$ git pull https://git.openjdk.java.net/jdk pull/3633/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3633`

View PR using the GUI difftool: \
`$ git pr show -t 3633`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3633.diff">https://git.openjdk.java.net/jdk/pull/3633.diff</a>

</details>

### CODE REVIEW

**Review by adinn — APPROVED** [review]
Looks good.

**Comment by dholmes-ora** [comment]
Can you please add comments to the new test explaining exactly what it is that the test is doing, why you need to use a jasm file etc. Also add an @bug tag to the test.

Thanks,
David

File: test/hotspot/jtreg/runtime/LocalVariableTable/TestLongLVTHelper.jasm [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * questions.
+ */
+
+package runtime/LocalVariableTable;
+
+public class  TestLongLVTHelper
+{
+
+public static Method tst:"()I"
+    stack 3 locals 30000
```
Comment (theRealAph) [inline test/hotspot/jtreg/runtime/LocalVariableTable/TestLongLVTHelper.jasm:31]: Looks good. You might as well make locals = 65535 for a better stress test.

Comment (voitylov) [reply test/hotspot/jtreg/runtime/LocalVariableTable/TestLongLVTHelper.jasm]: Sure, done.

**Comment by voitylov** [comment]
@dholmes-ora @theRealAph thanks for your comments. Please check the updated version.

**Comment by adinn** [comment]
@voitylov The patch still looks good modulo one niggle I have with the comment.

You use the phrase "local variable table" (twice). That jarred when I read it because it's the term used for the debug mode bytecode attribute that records names and liveness ranges for parameter & local variables found in the java source. It would avoid any potential for confusion if you used a different term, e.g. "local vars stack region".

**Review by theRealAph — APPROVED** [review]

**Comment by voitylov** [comment]
@adinn thank you for noticing. I updated the terminology and renamed the test.

**Comment by adinn** [comment]
@voitylov thanks looks good.

**Review by adinn — APPROVED** [review]

**Comment by adinn** [comment]
@voitylov When you are ready type the /integrate command in a new comment and I will happily sponsor the change.

**Comment by voitylov** [comment]
@dholmes-ora do you have any further comments?

**Comment by dholmes-ora** [comment]
Nothing further form me (I was away for a few days) - the test comments look good. I did not actually review the change itself (not my area).

Thanks,
David

**Comment by voitylov** [comment]
@adinn @theRealAph thank you for review, @dholmes-ora thank you for your comments!

**Pull request closed without merging**